### DISCUSSION TITLE
How to create a custom column which is a function of other results?

### DISCUSSION THREAD

**Discussion by quixoticaxis** [dotnet/BenchmarkDotNet#2443] [Q&A] [answered]
Hello.

I'm trying to implement a simple custom column that would contain `Mean / N` value where mean is a mean time from the standard result attributes, and `N` is a custom parameter that controls the number of iterations inside of my benchmark.
I understand how to get the value of `N` for a custom column, but I can't seem to find how to get the `Mean` value.

**Comment by leonvandermeer** [comment, marked as answer]
In the `GetValue` implementation of your column, you can get to the the parameter's value and benchmark's statistics like so:
```
public string GetValue(Summary summary, BenchmarkCase benchmarkCase, SummaryStyle style) {
  int n = (int)benchmarkCase.Parameters["N"];
  Statistics statistics = summary[benchmarkCase]!.ResultStatistics!;
  double mean = TimeInterval.FromNanoseconds(statistics.Mean).ToSeconds();
  double result = mean / n;
  return string.Create(style.CultureInfo, $"{result:F3} s");
}
```

In another column, you can calculate the error, assuming the error of N is zero:
````
double result = ...; double mean = ...; as above
double meanStdDev = TimeInterval.FromNanoseconds(statistics.StandardDeviation).ToSeconds();
double resultStdDev = result * meanStdDev / mean;
return string.Create(style.CultureInfo, $"{resultStdDev:F3} s");
````

**Reply by quixoticaxis** [reply]
Thank you.
